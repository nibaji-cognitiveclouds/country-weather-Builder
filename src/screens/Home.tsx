/** @format */

import React from "react";
import { View, TextInput, Button } from "react-native";
import { useNavigation } from "@react-navigation/native";

import { HomeStyle } from "../styles/home";
import { StackNavigationProp } from "@react-navigation/stack";
import { RootStackParamList } from "../types/navigation";

const Home: React.FC = () => {
	const [text, setText] = React.useState<string>("");

	const navigation = useNavigation<StackNavigationProp<RootStackParamList>>();

	return (
		<View style={HomeStyle.page}>
			<TextInput
				placeholder="Enter country"
				placeholderTextColor={"grey"}
				style={HomeStyle.textInput}
				onChangeText={(text) => {
					setText(text);
				}}
			/>

			<Button
				disabled={text?.length == 0}
				title="Search"
				onPress={() => {
					navigation.navigate("List", { text: text });
				}}
			/>
		</View>
	);
};

export default Home;
