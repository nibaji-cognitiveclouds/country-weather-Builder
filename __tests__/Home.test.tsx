/** @format */

import { render } from "@testing-library/react-native";
import React from "react";
import Home from "../src/screens/Home";

jest.mock("@react-navigation/native", () => {
	return {
		useNavigation: jest.fn(),
	};
});

describe("Home", () => {
	it("renders without crash", () => {
		render(<Home />);
	});

	it("matches Snapshot", () => {
		const tree = render(<Home />);
		expect(tree.toJSON()).toMatchSnapshot();
	});

	it("contains component with req testID", () => {
		const tree = render(<Home />);
		expect(tree.findByTestId("home")).toBeTruthy();
	});
});
