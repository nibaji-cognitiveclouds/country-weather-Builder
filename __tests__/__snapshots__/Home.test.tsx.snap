// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches Snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="home"
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Enter country name"
    placeholderTextColor="grey"
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "grey",
        "borderRadius": 5,
        "borderWidth": 1,
        "marginBottom": 10,
        "padding": 5,
        "width": "80%",
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {},
        ]
      }
    >
      <Text
        disabled={true}
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#cdcdcd",
            },
          ]
        }
      >
        Submit
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
